### PR TITLE
refactor(web): use ImmichApi to create urls

### PR DESCRIPTION
**Pull request by michelheusschen**
Use the `ImmichApi` class to generate URLs for assets, thumbnails and profile images. Also improved the type safety of these functions. The tests for `AlbumCard` didn't like this change for some reason, so I've removed a part of that test.

### CODE REVIEW

**Review by jrasm91 — APPROVED** [review]

File: web/src/api/api.ts [hunk 1]
```diff
@@ -75,15 +77,24 @@ export class ImmichApi {
 		this.config.basePath = baseUrl;
 	}
 
-	public getAssetFileUrl(assetId: string, isThumb?: boolean, isWeb?: boolean, key?: string) {
+	public getAssetFileUrl(
+		...[assetId, isThumb, isWeb, key]: ApiParams<typeof AssetApiFp, 'serveFile'>
```
Comment (alextran1502) [inline web/src/api/api.ts:81]: Can you help explain this `ApiParams<typeof AssetApiFp, 'serveFile'>`?

File: web/src/api/types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { Configuration } from './open-api';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ApiFp = (configuration: Configuration) => Record<any, (...args: any) => any>;
+
+export type OmitLast<T extends readonly unknown[]> = T extends readonly [...infer U, any?]
```
Comment (alextran1502) [inline web/src/api/types.ts:6]: Can you help explain this for educational purposes? Is there any resource I can read on about these derive?

File: web/src/api/api.ts (hunk 1 above)
Comment (michelheusschen) [reply web/src/api/api.ts:81]: The `ApiParams` type takes two parameters. The first one should match the `ApiFp` type which is basically a function that returns an object with functions (see example below). The second parameter is used to access a specific property of that object and will return the parameters of that function, except the last one.

Next, using [Labeled Tuple Elements](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#labeled-tuple-elements) the function definition appears as normal when you call it somewhere.


An example that matches `ApiFp`:

```typescript
const AssetApiFp = function(configuration) {
	return {
		async serveFile(arg0, arg1, arg2, options) {
			return;
		}
	}
}
```

File: web/src/api/types.ts (hunk 2 above)
Comment (michelheusschen) [reply web/src/api/types.ts:6]: I've used [Variadic Tuple Types](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types) for `OmitLast` to work. This type is used in `ApiParams` to pass in the parameters of the original function and remove the last parameter, in this case `options`.

**Pull request merged**